sqlglot/dialects/bigquery.py

```diff
@@ -474,11 +474,29 @@ def _parse_table_parts(
                 if rest and this:
                     this = exp.Dot.build([this, *rest])  # type: ignore
 
-                table = exp.Table(this=this, db=db, catalog=catalog)
+                table = exp.Table(
+                    this=this, db=db, catalog=catalog, pivots=table.args.get("pivots")
+                )
                 table.meta["quoted_table"] = True
 
             return table
 
+        def _parse_column(self) -> t.Optional[exp.Expression]:
+            column = super()._parse_column()
+            if isinstance(column, exp.Column) and any("." in p.name for p in column.parts):
```

Comment:
parts is called twice here maybe let's store it in a var because it's not free, can change for table too

Reply:
good catch, missed it

Reply:
https://github.com/tobymao/sqlglot/pull/3330/commits/f77f064e33d833995f6ea900ead1dc680f3f62bc